qiskit/synthesis/evolution/product_formula.py

```diff
@@ -108,8 +114,50 @@ def wrap_atomic_evolution(output, operator, time):
         else:
             self.atomic_evolution = atomic_evolution
 
+    def expand(
+        self, evolution: PauliEvolutionGate
+    ) -> list[tuple[str, tuple[int], ParameterValueType]]:
+        """Apply the product formula to expand the Hamiltonian in the evolution gate.
+
+        Args:
+            evolution: The :class:`.PauliEvolutionGate`, whose Hamiltonian we expand.
+
+        Returns:
+            A list of Pauli rotations in a sparse format, where each element is
+            ``(paulistring, qubits, coefficient)``. For example, the Lie-Trotter expansion
+            of ``H = XI + ZZ`` would return ``[("X", [1], 1), ("ZZ", [0, 1], 1)]``.
+        """
+        raise NotImplementedError(
+            f"The method ``expand`` is not implemented for {self.__class__}. Implement it to "
+            f"automatically enable the call to {self.__class__}.synthesize."
+        )
```

Comment:
Can we move this abstract method to its parent class `EvolutionSynthesis`?

Reply:
Only `ProductFormula`s can expand, general `EvolutionSynthesis` can do arbitrary things to implement the evolution (such as computing the matrix evo) and don't have `expand` 🙂